src/cogent3/core/new_alignment.py

```diff
@@ -0,0 +1,449 @@
+            self.get_seq_array(seqid=seqid, start=start, stop=stop)
+        )
+
+    def get_seq_bytes(
+        self, *, seqid: str, start: int = None, stop: int = None
+    ) -> bytes:
+        return self.get_seq_str(seqid=seqid, start=start, stop=stop).encode("utf8")
+
+    def iter_names(self, *, name_order: tuple[str] = None) -> Iterator:
+        yield from process_name_order(self._name_order, name_order)
+
+    def get_seq_view(self, seqid: str) -> SeqDataView:
+        seq_len = len(self._data[seqid])
+        return SeqDataView(seq=self, seqid=seqid, seq_len=seq_len)
+
+    def iter_seq_view(self, *, name_order: tuple[str] = None) -> Iterator:
+        # Should this output SeqView or SeqDataView?
+        seqids = process_name_order(self._name_order, name_order)
+        for seqid in seqids:
+            yield self.get_seq_view(seqid=seqid)
+
+
+@singledispatch
+def seq_to_gap_coords(
+    seq: Union[str, numpy.ndarray], moltype: MolType
+) -> tuple[str, IndelMap]:
+    """
+    Takes a sequence with (or without) gaps and returns an ungapped sequence
+    and records the position and length of gaps in the original parent sequence
+    """
+    raise NotImplementedError(f"{seq} not implemented for type {type(seq)}")
+
+
+@seq_to_gap_coords.register
+def _(seq: str, moltype: MolType) -> tuple[str, IndelMap]:
+    seq = moltype.make_seq(seq)
+    gap_char = moltype.gap
+    indel_map, ungapped_seq = seq.parse_out_gaps()
+
+    if indel_map.num_gaps == 0:
+        return str(ungapped_seq), numpy.array([], dtype=int)
+
+    return str(ungapped_seq), indel_map
+
+
+@seq_to_gap_coords.register
+def _(seq: numpy.ndarray, moltype: MolType) -> tuple[str, IndelMap]:
+    gap_char = moltype.alphabets.degen_gapped.index(moltype.gap)
+    # Create boolean array of gaps to get ungapped
+    gaps_bool = seq == gap_char
+    ungapped = seq[~gaps_bool]
+
+    # no gaps in seq
+    if numpy.array_equal(ungapped, seq):
+        return ungapped, numpy.array([], dtype=int)
+
+    parent_len = len(seq)
+    in_gap = False
+    parent_coords = []
+    start = 0
+    for i, gapped in enumerate(gaps_bool):
+        if gapped and not in_gap:
+            start = i
+            in_gap = True
+        if not gapped and in_gap:
+            parent_coords.append([start, i])
+            in_gap = False
+        if gapped and i == parent_len - 1:
+            # End of the sequence
+            parent_coords.append([start, i + 1])
+
+    # Format for IndelMap
+    parent_coords = numpy.array(parent_coords)
+    gap_lengths = numpy.array([end - start for start, end in parent_coords])
+    cum_lengths = gap_lengths.cumsum()
+    # Get gap start positions in sequence coords
+    gap_pos = parent_coords.T[0] - numpy.append(0, cum_lengths[:-1, numpy.newaxis])
+
+    indel_map = IndelMap(
+        gap_pos=gap_pos, cum_gap_lengths=cum_lengths, parent_length=parent_len
+    )
+
+    return ungapped, indel_map
+
+
+class AlignedDataView(SeqDataView):
```

Comment:
I think this should inherit from `SliceRecordABC`, but not confident

Reply:
Yep I think so too, but leaving AlignedData/View until SeqData is (mostly) finished

Reply:
Agree, we don't worry about Aligned stuff yet.